x-pack/plugins/cloud_security_posture/public/components/page_template.tsx

```diff
@@ -45,28 +51,46 @@ const defaultProps: KibanaPageTemplateProps = {
   template: 'default',
 };
 
-interface CspPageTemplateProps extends KibanaPageTemplateProps {
-  isLoading?: boolean;
-  loadingText?: string;
-}
+const NO_DATA_CONFIG: KibanaPageTemplateProps['noDataConfig'] = {
+  pageTitle: NO_DATA_CONFIG_TITLE,
+  solution: NO_DATA_CONFIG_SOLUTION_NAME,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {
+    elasticAgent: {
+      // TODO: Use `href` prop to link to our own integration once we have it
+      title: NO_DATA_CONFIG_BUTTON,
+      description: NO_DATA_CONFIG_DESCRIPTION,
+    },
+  },
+};
+
+export const CspPageTemplate: React.FC<KibanaPageTemplateProps> = ({ children, ...props }) => {
+  // TODO: Consider using more sophisticated logic to find out if our integration is installed
+  const kubeBeatQuery = useKubebeatDataView();
+
+  let noDataConfig: KibanaPageTemplateProps['noDataConfig'];
+  if (kubeBeatQuery.status === 'success' && !kubeBeatQuery.data) {
+    noDataConfig = NO_DATA_CONFIG;
+  }
 
-export const CspPageTemplate: React.FC<CspPageTemplateProps> = ({
-  children,
-  isLoading,
-  loadingText = LOADING,
-  ...props
-}) => {
   return (
-    <KibanaPageTemplate {...defaultProps} {...props}>
+    <KibanaPageTemplate {...DEFAULT_PROPS} {...props} noDataConfig={noDataConfig}>
       <EuiErrorBoundary>
-        {isLoading ? (
-          <>
-            <EuiSpacer size="xxl" />
-            <CspLoadingState>{loadingText}</CspLoadingState>
-          </>
-        ) : (
-          children
+        {kubeBeatQuery.status === 'loading' && <CspLoadingState>{LOADING}</CspLoadingState>}
+        {kubeBeatQuery.status === 'error' && (
+          <EuiEmptyPrompt
+            color="danger"
+            iconType="alert"
+            title={
+              <EuiTitle>
+                <h2>{ERROR_LOADING_DATA}</h2>
+              </EuiTitle>
+            }
+          />
         )}
+        {kubeBeatQuery.status === 'success' && children}
```

Comment:
kinda nit: since selecting which state to render is an `OR` and not an `AND` I would opt to use a function with an `if` statement that returns the matching component.

```
const render = (state, children)  => {
  if (state === "loading") return <CspLoadingState />
  // rest of states..
  
  return children
}
```

Reply:
IMO this is fine when checking different values of a single property. This is a very common React pattern